Blockzilla/AddSearchEngineViewController.swift

```diff
@@ -142,25 +144,38 @@ class AddSearchEngineViewController: UIViewController, UITextViewDelegate {
         guard let name = nameInput.text else { return }
         guard let template = templateInput.text else { return }
         
+        if !AddSearchEngineViewController.isValidTemplate(template) || !searchEngineManager.isValidSearchEngineName(name) {
+            Toast(text: UIConstants.strings.errorTryAgain).show()
+            return
+        }
+        
         delegate.addSearchEngineViewController(self, name: name, searchTemplate: template)
         Toast(text: UIConstants.strings.NewSearchEngineAdded).show()
         self.navigationController?.popViewController(animated: true)
     }
     
     func textViewDidBeginEditing(_ textView: UITextView) {
-        templatePlaceholderLabel.isHidden = !isNullOrEmpty(text: textView.text)
+        templatePlaceholderLabel.isHidden = !textView.text.isEmpty
     }
     
     func textViewDidChange(_ textView: UITextView) {
-        templatePlaceholderLabel.isHidden = !isNullOrEmpty(text: textView.text)
+        templatePlaceholderLabel.isHidden = !textView.text.isEmpty
     }
     
     func textViewDidEndEditing(_ textView: UITextView) {
-        templatePlaceholderLabel.isHidden = !isNullOrEmpty(text: textView.text)
+        templatePlaceholderLabel.isHidden = !textView.text.isEmpty
     }
     
-    private func isNullOrEmpty(text:String?) -> Bool{
-        guard let text = text else { return true }
-        return text.isEmpty
+    static func isValidTemplate(_ template:String) -> Bool {
+        if template.isEmpty {
+            return false
+        }
+        
+        if !template.contains("%s") {
+            return false
+        }
+        
+        guard let url = URL(string: template.addingPercentEncoding(withAllowedCharacters: CharacterSet.urlFragmentAllowed)!) else { return false }
+        return url.isWebPage()
```

Comment:
@joeyg @boek @bkmunar I for some reason can't find the API doc on this - what does this call do?

Reply:
@Sdaswani https://github.com/mozilla-mobile/focus-ios/blob/master/Blockzilla/OverlayView.swift#L211

Reply:
Ok I think this patch set is fairly straighforward, and it has tests. One more thing to do is add the HTTP 200 check, but it needs to be done in a way that doesn't block the UI and it needs to time out in a reasonable fashion. Don't stress too much about 'no internet' - we'll do that later. 

@joeyg @boek can one of you add that?

So when the user presses 'Save', check it's a valid URL (per all the checks here), and that it resolves to a HTTP 200.

Reply:
Or @boek if you want ot do something more targeted for https://github.com/mozilla-mobile/focus-ios/issues/795, that's fine too.